src/tools/compiletest/src/load_cfg/itemlist.rs

```diff
@@ -0,0 +1,733 @@
+                    };
+                    let content = match_any(
+                        |matcher, pass| matcher.try_match_directive(content, pass),
+                        make_err,
+                    )?
+                    .ok_or_else(|| make_err().unwrap_err())?;
+
+                    *item = Item {
+                        val: ItemVal::RevisionSpecificExpanded { revs, content: Box::new(content) },
+                        pos: item.pos,
+                        used: Cell::new(false),
+                    };
+                }
+                ItemVal::RevisionSpecificExpanded { .. } => unreachable!(
+                    "if this visit has never happened before, there should be \
+                    no expanded nodes."
+                ),
+                _ => (),
+            }
+        }
+
+        Ok(())
+    }
+}
+
+// TODO: parse revisions into `ItemVal`s.
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ItemVal<'src> {
+    /* Flags that can only be set to true */
+    BuildPass,
+    BuildFail,
+    CheckPass,
+    CheckFail,
+    RunPass,
+    RunFail,
+    NoPreferDynamic,
+    NoAutoCheckCfg,
+
+    /* Boolean flags that can be set with `foo-bar` or `no-foo-bar` */
+    ShouldIce(bool),
+    ShouldFail(bool),
+    BuildAuxDocs(bool),
+    ForceHost(bool),
+    CheckStdout(bool),
+    CheckRunResults(bool),
+    DontCheckCompilerStdout(bool),
+    DontCheckCompilerStderr(bool),
+    PrettyExpanded(bool),
+    PrettyCompareOnly(bool),
+    CheckTestLineNumbersMatch(bool),
+    StderrPerBitwidth(bool),
+    Incremental(bool),
+    DontCheckFailureStatus(bool),
+    RunRustfix(bool),
+    RustfixOnlyMachineApplicable(bool),
+    CompareOutputLinesBySubset(bool),
+    KnownBug(bool),
+    RemapSrcBase(bool),
+
+    /* Key-value directivesthat can be set with `foo-bar: baz qux` */
+    ErrorPattern(&'src str),
+    RegexErrorPattern(&'src str),
+    CompileFlags(&'src str),
+    Edition(&'src str),
+    RunFlags(&'src str),
+    PrettyMode(&'src str),
+    AuxBin(&'src str),
+    AuxBuild(&'src str),
+    AuxCrate(&'src str),
+    AuxCodegenBackend(&'src str),
+    ExecEnv(&'src str),
+    UnsetExecEnv(&'src str),
+    RustcEnv(&'src str),
+    UnsetRustcEnv(&'src str),
+    ForbidOutput(&'src str),
+    FailureStatus(&'src str),
+    AssemblyOutput(&'src str),
+    TestMirPass(&'src str),
+    LlvmCovFlags(&'src str),
+    FilecheckFlags(&'src str),
+    /// The revisions list
+    Revisions(&'src str),
+
+    /* Directives that are single keys but have variable suffixes */
+    /// Ignore something specific about this test.
+    ///
+    /// `//@ ignore-x86`
+    Ignore {
+        what: &'src str,
+    },
+    /// Components needed by the test
+    ///
+    /// `//@ needs-rust-lld`
+    Needs {
+        what: &'src str,
+    },
+    /// Run only if conditions are met
+    ///
+    /// `//@ only-linux`
+    Only {
+        what: &'src str,
+    },
```

Comment:
Suggestion (not necessarily for this PR): `only-*` and `ignore-*` are very tricky (if we look at `header/cfg.rs`) in that the suffixes are parsed with a sequence of precedence-based matchers. If you specify e.g. `ignore-x86`, you have to look and reason *really* hard to realize "oh it's the `target_cfg.all_archs` that's matching and taking effect". We probably want to eventually straight up ask for a longer prefix to help make it excruciatingly clear what we're trying to match on (this would actually better match `ui_test`)

```
//@ only-arch-x86
```